src/Integration.Vsix/SLCore/SLCoreEmbeddedPluginJarLocator.cs

```diff
@@ -55,4 +64,36 @@ public List<string> ListJarFiles()
         }
         return new List<string>();
     }
+
+    public Dictionary<string, string> ListConnectedModeEmbeddedPluginPathsByKey()
+    {
+        var connectedModeEmbeddedPluginPathsByKey = new Dictionary<string, string>();
+        var embeddedPluginFilePaths = ListJarFiles();
+
+        foreach (var kvp in connectedModePluginNamePatternByPluginKey)
+        {
+            if (GetPathByPluginKey(embeddedPluginFilePaths, kvp.Key, kvp.Value) is {} pluginFilePath)
+            {
+                connectedModeEmbeddedPluginPathsByKey.Add(kvp.Key, pluginFilePath);
+            }
+        }
+
+        return connectedModeEmbeddedPluginPathsByKey;
+    }
+
+    private string GetPathByPluginKey(List<string> pluginFilePaths, string pluginKey, string pluginNameRegexPattern)
+    {
+        var regex = new Regex(pluginNameRegexPattern);
+        var matchedFilePaths = pluginFilePaths.Where(jar => regex.IsMatch(jar)).ToList();
+        switch (matchedFilePaths.Count)
+        {
+            case 0:
+                logger.LogVerbose(Strings.ConnectedModeEmbeddedPluginJarLocator_JarsNotFound);
+                break;
+            case > 1:
+                logger.LogVerbose(Strings.ConnectedModeEmbeddedPluginJarLocator_MultipleJars, pluginKey);
+                break;
+        }
+        return matchedFilePaths.FirstOrDefault();
```

Comment:
This will return null for the `0` case.

Reply:
That is intended. Therefore no plugin will be added